x/revenue/keeper/evm_hooks.go

```diff
@@ -81,17 +81,16 @@ func (k Keeper) PostTxProcessing(
 		)
 	}
 
-	ctx.EventManager().EmitEvents(
-		sdk.Events{
-			sdk.NewEvent(
-				types.EventTypeDistributeDevRevenue,
-				sdk.NewAttribute(sdk.AttributeKeySender, msg.From().String()),
-				sdk.NewAttribute(types.AttributeKeyContract, contract.String()),
-				sdk.NewAttribute(types.AttributeKeyWithdrawerAddress, withdrawer.String()),
-				sdk.NewAttribute(sdk.AttributeKeyAmount, developerFee.String()),
-			),
-		},
-	)
+	err = ctx.EventManager().EmitTypedEvent(&types.EventDistributeRevenue{
+		Sender:            msg.From().String(),
+		Contract:          contract.String(),
+		WithdrawerAddress: withdrawer.String(),
+		Amount:            developerFee.String(),
+	})
+
+	if err != nil {
+		return err
+	}
```

Comment:
Returning an error will roll back the transaction. 
I think we should log error instead